pkg/registry/core/shoot/strategy.go

```diff
@@ -167,6 +169,25 @@ func (shootStrategy) Validate(ctx context.Context, obj runtime.Object) field.Err
 }
 
 func (shootStrategy) Canonicalize(obj runtime.Object) {
+	shoot := obj.(*core.Shoot)
+	if versionutils.ConstraintK8sGreaterEqual125.Check(semver.MustParse(shoot.Spec.Kubernetes.Version)) {
+		cleanupAdmissionPlugins(shoot)
+	}
+}
+
+func cleanupAdmissionPlugins(shoot *core.Shoot) {
+	var (
+		admissionPlugins      []core.AdmissionPlugin
+		shootAdmissionPlugins = shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins
+	)
+
+	for _, plugin := range shootAdmissionPlugins {
+		if plugin.Name != "PodSecurityPolicy" {
```

Comment:
Nit: Should we only remove it if it is disabled? This is enforced implicitly I think because otherwise we could not reach this code, but still?

Reply:
In `1.25`+, we cannot enable it as well.